code/modules/projectiles/projectile.dm

```diff
@@ -354,6 +355,7 @@
 		if(ammo.bullet_color)
 			set_light_color(ammo.bullet_color)
 			set_light_on(TRUE)
+			add_overlay(emissive_appearance(icon, icon_state, src, layer, attached = TRUE))
```

Comment:
don't set overlays outsid of update_overlays - update appearance can be flagged to specifically call that as needed.